### ISSUE TITLE
DatePicker fails accessibility tesitng. 

### ISSUE THREAD

**Issue by mhemesath** [cerner/terra-core#1068]
# Issue Description
Date picker fails accessibility testing

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
Passes accessibility tests

## Current Behavior
* Control button fails from not having text. Related to #1058 
* Contrast ratio of selected day is not high enough. I think this is because the text is so small.
* The date picker doesn't generate a label for its input. *Note: this may just be an example update, but looking at doc I don't see anything on how to create a datepicker with a label*

## Steps to Reproduce

1. Install [Axe Chrome Extension](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)
2. Navigate to a [date picker example](http://engineering.cerner.com/terra-core/#/tests/date-picker-tests/default) and open the datepicker by activating the control.
3. Run axe by going to developer tools and clicking axe tab.

## Environment

* Component Version:
* Browser Name and Version:
* Operating System and version (desktop or mobile):

**Comment by PaulSolDev** [comment]
Orion JIRA created

**Comment by jmsv6d** [comment]
> Control button fails from not having text. Related to #1058 

This was resolved with the uplift to use terra-button-2.0.0

> Contrast ratio of selected day is not high enough. I think this is because the text is so small.

@neilpfeiffer Needs ux input. The full message is: "Element has insufficient color contrast of 3.56 (foreground color: #ffffff, background color: #4285f4, font size: 9.8pt, font weight: normal). Expected contrast ratio of 4.5:1"

> The date picker doesn't generate a label for its input. Note: this may just be an example update, but looking at doc I don't see anything on how to create a datepicker with a label

I feel like we should just turn off this rule for the date picker instead of adding labels to each of the test examples. A date field component would be responsible with adding the correct accessibility attributes for a label, not the date picker itself. We will come across this issue as we add accessibility tests to our other field components.

**Comment by bjankord** [comment]
The [individual](https://github.com/aldavigdis) driving the [accessibility PR for react-datepicker](https://github.com/Hacker0x01/react-datepicker/pull/1408) is one of the wordpress contributors. They were hoping to improve the accessibility of the react-datepicker component as they use it in there new wordpress editor. I was hoping the PR would land soon and we could reap the benefits of that work, but it seems the the PR has stalled out. Recently, [the wordpress team switched](https://github.com/WordPress/gutenberg/pull/7621) from using react-datepicker (the third-party lib we use), to airbnb's react-dates component in an effort to improve their datepicker experience.

I've considered switching to [airbnb's react-dates component](https://github.com/airbnb/react-dates) as well as it appears to offer an improved accessible experience. I've talked with the other maintainers and there is a plan to do a few spikes on various options. 
* Spike on what effort it would take to migrate / make sure we don't paint ourselves into a corner if we switching to react-dates.
* Spike on improving the accessibility of Cerner's in-house datepicker (Sol) and wrapping that.
* Spike on if it's a viable option to remove the datepicker component from keyboard / assisted technology based workflows, leaving date entry as a text field.
